File: 1615-maximal-network-rank/1615-maximal-network-rank.py
class Solution:
    def maximalNetworkRank(self, n: int, roads: List[List[int]]) -> int:
        connected = [[False] * n for _ in range(n)]
        count = [0] * n

        for road in roads:
            count[road[0]] += 1
            count[road[1]] += 1
            connected[road[0]][road[1]] = True
            connected[road[1]][road[0]] = True

        res = 0

        for i in range(n):
            for j in range(i + 1, n):
                rank = count[i] + count[j] - (connected[i][j] == True)
                res = max(res, rank)

        return res

File: LeetCode/1615-maximal-network-rank/1615-maximal-network-rank.cpp
class Solution {
public:
    int maximalNetworkRank(int n, vector<vector<int>>& roads) {
        vector<vector<bool>> connected(n, vector<bool>(n, false));
        vector<int> count(n, 0);

        for (const auto& road : roads) {
            count[road[0]]++;
            count[road[1]]++;
            connected[road[0]][road[1]] = true;
            connected[road[1]][road[0]] = true;
        }

        int res = 0;

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int rank = count[i] + count[j] - (connected[i][j] ? 1 : 0);
                res = max(res, rank);
            }
        }

        return res;
    }
};